lib/makeCmdTasks.js

```diff
@@ -41,7 +41,7 @@ export const makeCmdTasks = async ({ commands, cwd, files, gitDir, shell, verbos
         )
       }
 
-      const task = resolveTaskFn({ command, cwd, files, gitDir, isFn, shell, verbose })
+      const task = resolveTaskFn({ command, cwd, files: [...files], gitDir, isFn, shell, verbose })
```

Comment:
Do you think these both places should make a new deep copy? Otherwise the cmd might still modify the list of files before it goes to the task resolving step.

Reply:
Not sure to understand what you mean, sorry

Reply:
Sorry, let me rephrase: your original fix looked something like this:

```js
const newFiles = [...files]

...

const resolved = isFn ? await cmd(newFiles) : cmd

...

const task = resolveTaskFn({ command, cwd, files: newFiles gitDir, isFn, shell, verbose })
```

While this prevented the mutation of `files`, it still passed `newFiles` to the `resolveTaskFn` which actually generates the executed command, and so `await cmd(newFiles)` might have mutated `newFiles` by making it empty, for example.

I took the liberty of adjusting the implementation a bit by passing `await cmd([...files])`, and then supply the original, unmutated `files` to `resolveTaskFn`. I also added a simple unit test.

Good job finding this bug. 👍